src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -262,14 +267,7 @@ internal Silo(SiloInitializationParameters initializationParams)
                     services.TryAddSingleton<ClientObserversPlacementDirector>();
                     
                     services.TryAddSingleton<Func<IGrainRuntime>>(sp => () => sp.GetRequiredService<IGrainRuntime>());
-                    if (usingCustomServiceProvider)
-                    {
-                        services.TryAddSingleton<GrainCreator>();
-                    }
-                    else
-                    {
-                        services.TryAddSingleton(sp => new GrainCreator(null, sp.GetRequiredService<Func<IGrainRuntime>>()));
-                    }
+                    services.TryAddSingleton<GrainCreator>();
```

Comment:
I made a note about this above (you can see @jdom's comment on this PR).
This is needed because some tests require access to things which were previously static but are now no longer exposed in a static manner. Therefore, we inject them into the grains used by the tests. The issue is that under the existing setup, constructor injection is not enabled for grains unless the user specifies a startup class - which is weird because even if we don't actually inject anything in our startup class, it enables dependency injection. It's a violation of the principle of least surprise.